src/helper/SM4T_translate.py

```diff
@@ -2,9 +2,317 @@
+    "ka": "kat",
+    "kk": "kaz",
+    "km": "khm",
+    "kn": "kan",
+    "ko": "kor",
+    "la": "lat",
+    "lb": "ltz",
+    "ln": "lin",
+    "lo": "lao",
+    "lt": "lit",
+    "lv": "lvs",
+    "mg": "mlg",
+    "mi": "mri",
+    "mk": "mkd",
+    "ml": "mal",
+    "mn": "khk",
+    "mr": "mar",
+    "ms": "zsm",
+    "mt": "mlt",
+    "my": "mya",
+    "ne": "npi",
+    "nl": "nld",
+    "nn": "nno",
+    "no": "nob",
+    "oc": "oci",
+    "pa": "pan",
+    "pl": "pol",
+    "ps": "pbt",
+    "pt": "por",
+    "ro": "ron",
+    "ru": "rus",
+    "sa": "san",
+    "sd": "snd",
+    "si": "sin",
+    "sk": "slk",
+    "sl": "slv",
+    "sn": "sna",
+    "so": "som",
+    "sq": "sqi",
+    "sr": "srp",
+    "su": "sun",
+    "sv": "swe",
+    "sw": "swh",
+    "ta": "tam",
+    "te": "tel",
+    "tg": "tgk",
+    "th": "tha",
+    "tk": "tuk",
+    "tl": "tgl",
+    "tr": "tur",
+    "tt": "tat",
+    "uk": "ukr",
+    "ur": "urd",
+    "uz": "uzn",
+    "vi": "vie",
+    "yi": "yid",
+    "yo": "yor",
+    "zh": "cmn",
+    "afr": "afr",
+    "amh": "amh",
+    "arb": "arb",
+    "ary": "ary",
+    "arz": "arz",
+    "asm": "asm",
+    "azj": "azj",
+    "bel": "bel",
+    "ben": "ben",
+    "bos": "bos",
+    "bul": "bul",
+    "cat": "cat",
+    "ceb": "ceb",
+    "ces": "ces",
+    "ckb": "ckb",
+    "cmn": "cmn",
+    "cym": "cym",
+    "dan": "dan",
+    "deu": "deu",
+    "ell": "ell",
+    "eng": "eng",
+    "est": "est",
+    "eus": "eus",
+    "fin": "fin",
+    "fra": "fra",
+    "fuv": "fuv",
+    "gaz": "gaz",
+    "gle": "gle",
+    "glg": "glg",
+    "guj": "guj",
+    "heb": "heb",
+    "hin": "hin",
+    "hrv": "hrv",
+    "hun": "hun",
+    "hye": "hye",
+    "ibo": "ibo",
+    "ind": "ind",
+    "isl": "isl",
+    "ita": "ita",
+    "jav": "jav",
+    "jpn": "jpn",
+    "kan": "kan",
+    "kat": "kat",
+    "kaz": "kaz",
+    "khk": "khk",
+    "khm": "khm",
+    "kir": "kir",
+    "kor": "kor",
+    "lao": "lao",
+    "lit": "lit",
+    "lug": "lug",
+    "luo": "luo",
+    "lvs": "lvs",
+    "mai": "mai",
+    "mal": "mal",
+    "mar": "mar",
+    "mkd": "mkd",
+    "mlt": "mlt",
+    "mni": "mni",
+    "mya": "mya",
+    "nld": "nld",
+    "nno": "nno",
+    "nob": "nob",
+    "npi": "npi",
+    "nya": "nya",
+    "ory": "ory",
+    "pan": "pan",
+    "pbt": "pbt",
+    "pes": "pes",
+    "pol": "pol",
+    "por": "por",
+    "ron": "ron",
+    "rus": "rus",
+    "slk": "slk",
+    "slv": "slv",
+    "sna": "sna",
+    "snd": "snd",
+    "som": "som",
+    "spa": "spa",
+    "srp": "srp",
+    "swe": "swe",
+    "swh": "swh",
+    "tam": "tam",
+    "tel": "tel",
+    "tgk": "tgk",
+    "tgl": "tgl",
+    "tha": "tha",
+    "tur": "tur",
+    "ukr": "ukr",
+    "urd": "urd",
+    "uzn": "uzn",
+    "vie": "vie",
+    "yor": "yor",
+    "yue": "yue",
+    "zsm": "zsm",
+    "zul": "zul",
+}
+
+
+def check_language_supported(language):
+    language = language_map[language] if language in language_map else language
```

Comment:
https://docs.python.org/3/library/stdtypes.html#dict.get

Reply:
So this confuses me a bit: is the `else language` case always invalid? Because it seems that way in line 331/334.

Reply:
great catch, it would have still worked, but it wasn't actually necessary to keep the code that way.

Reply:
Okay then the business logic is not a problem and does not need to be changed.

However using languagemap.get(language, language) seems more concise.